src/main/kotlin/eu/europa/ec/eudi/pidissuer/adapter/out/persistence/InMemoryDeferredCredentialRepository.kt

```diff
@@ -27,7 +28,8 @@ import org.slf4j.LoggerFactory
 
 private val log = LoggerFactory.getLogger(InMemoryDeferredCredentialRepository::class.java)
 class InMemoryDeferredCredentialRepository(
-    private val data: MutableMap<TransactionId, CredentialResponse.Issued<JsonElement>?> = mutableMapOf(),
+    private val data: MutableMap<TransactionId, Pair<CredentialResponse.Issued<JsonElement>, RequestedResponseEncryption>?> =
```

Comment:
The nullable pair is not correct.  
`RequestedResponseEncryption` is nullable on its own.
Perhaps a (not-null) pair of nullables or even better a dedicated class  

```kotlin
data class DefPersist(
   val enc: RequestedResponseEncryption?,
   val issued: CredentialResponse.Issued<JsonElement>?
)

```